server/test/api.js

```diff
@@ -0,0 +1,26 @@
+const chai = require("chai")
+const { expect, assert } = require("chai")
+const chaiHttp = require("chai-http")
+
+chai.use(require('chai-json'))
+chai.use(chaiHttp)
+
+describe("Testing API calls", () => {
+    it("Should receive {\"first\": \"John\", \"last\": \"Doe\"} after POST to /api/test", () => {
+        const dummyData = {
+            "first": "John",
+            "last": "Doe",
+        }
+
+        chai.request("http://localhost:4000")
+            .post("/api/test")
+            .send({
```

Comment:
I'm curious as to the syntax of javascript. Is this running `chai.request()` followed by `chai.post()` and `chai.send()`? or is this instead read as `chai.request().post().send()`? The second one seems like it would be the case to me, but I think javascript handles new lines a lot like python so I could be wrong.

Reply:
I believe it's the second. Under the hood it might be doing:
- chai.request() -> Request object r
- r.post() -> Set parameter of r's request type to POST
- r.send() -> Set data to be sent by Request r